llvm/lib/CodeGen/AsmPrinter/AccelTable.cpp

```diff
@@ -467,26 +427,43 @@ getFormForIdxParent(const DenseSet<OffsetAndUnitID> &IndexedOffsets,
   return dwarf::Form::DW_FORM_flag_present;
 }
 
+void DebugNamesAbbrev::Profile(FoldingSetNodeID &ID) const {
+  ID.AddInteger(DieTag);
+  for (const DebugNamesAbbrev::AttributeEncoding &Enc : AttrVect) {
+    ID.AddInteger(Enc.Index);
+    if (Enc.Index == dwarf::DW_IDX_parent)
+      ID.AddInteger(Enc.Form);
```

Comment:
For DW_IDX_parent it can be DW_FORM_flag_present or DW_FORM_ref4. It depends on a DIE. For all others form is unique per module.
For DW_IDX_die_offset it's always DW_IDX_die_offset.
For DW_IDX_compile_unit/DW_IDX_type_unit it can change based on size of CU/TU lists in the module. So DW_FORM_data{1,2,4}.

Reply:
Oh, but it's the same form for all entries, even the smaller ones - if the list is large enough?
Might be good to just include the form in the hash here, as we might optimize it later to use smaller forms for smaller entries? (at least we should assert the expected form so we can catch this later - but some amount of generality might just be easier to do now rather than assert and fix later)

Reply:
OK. Changed to always include form.